### PR TITLE
RELATED: RAIL-657 Added factory method for multiple instances

### PR DESCRIPTION
**Pull request by vojty**
`gooddata@5.0.0-vojty-factory-method-2018-03-01T09-38-56-648Z`

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/142/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/100/ : FAILURE in 1m 54s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/143/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/101/ : FAILURE in 2m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/144/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/102/ : SUCCESS in 2m 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/145/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/103/ : SUCCESS in 1m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/147/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/106/ : SUCCESS in 1m 47s

File: index.d.ts [hunk 1]
```diff
@@ -352,34 +352,46 @@ export interface IColor {
...
+    xhr: IXhr;
+    project: IProject;
+    execution: IExecution;
+    catalogue: ICatalogue;
+    config: IConfig;
+    md: IMetadata;
+    user: IUser;
+}
+
+export declare function factory(domain?: string): ISdk;
```
Comment (vaclavbohac) [inline index.d.ts]: ISdkOptions would be more suitable as parameter

File: src/gooddata.js [hunk 2]
```diff
@@ -25,7 +25,38 @@ import admin from './admin';
  * @module sdk
  * @class sdk
  */
-const gooddata = { config, xhr, user, md, execution, project, catalogue, admin };
-export default gooddata;
-module.exports = gooddata;
+// const gooddata = { config, xhr, user, md, execution, project, catalogue, admin };
```
Comment (vaclavbohac) [inline src/gooddata.js]: Please remove forgotten code

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/148/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/107/ : FAILURE in 1m 50s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/149/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/108/ : FAILURE in 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/150/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/109/ : SUCCESS in 1m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/151/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/110/ : SUCCESS in 2m 15s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/152/ : SUCCESS in 10s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/111/ : SUCCESS in 2m 00s

**Comment by zbycz** [comment]
`gooddata@5.0.0-pavelzgdc-vojty-factory-method-2018-02-23T12-12-05-193Z`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/153/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/112/ : SUCCESS in 2m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/158/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/118/ : SUCCESS in 3m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-js-opensource-repository-check/159/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/119/ : SUCCESS in 1m 56s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-js-merge-compliance-check/55/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-js-unit-tests-zuul-docker-el7/120/ : SUCCESS in 45s

**Pull request merged**